zerver/lib/stream_subscription.py

```diff
@@ -350,6 +350,7 @@ def get_subscriptions_for_send_message(
         | (Q(push_notifications=None) & Q(user_profile__enable_stream_push_notifications=True))
         | Q(email_notifications=True)
         | (Q(email_notifications=None) & Q(user_profile__enable_stream_email_notifications=True))
+        | Q(user_profile__enable_followed_topic_email_notifications=True)
```

Comment:
So I think in a world where we set `enable_followed_topic_email_notifications` to true by default, this line will defeat the performance benefits of this function, documented in its docstring. 

I think the fix for that problem is that we need to instead include every user who actually follows this topic in who we fetch these data for. 

Possibly a good way to do that would be for `get_recipient_info` to be responsible for this.

Reply:
> this line will defeat the performance benefits of this function, documented in its docstring.

Agreed.

> we need to instead include every user who actually follows this topic in who we fetch these data for.

Makes sense.
I have updated the logic.

> Possibly a good way to do that would be for get_recipient_info to be responsible for this.

We can do this, but it will lead to an increase in query count by 1.
Let me know, if the current approach has any downsides, we can change this accordingly.